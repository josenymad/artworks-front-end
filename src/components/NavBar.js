import React from "react";
import "../styles/nav-bar.css";
import { Link, useLocation } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";

const NavBar = ({
  searchQuery,
  handleSearchInput,
  handleSearchSubmit,
  signOut,
}) => {
  const { pathname } = useLocation();

  return (
    <nav className="navbar">
      <ul className="navbar__list">
        <li className="navbar__list-item">
          <Link to="/" className="navbar__list__link">
            All Artworks
          </Link>
        </li>
        <li className="navbar__list-item">
          <Link to="add-artwork" className="navbar__list__link">
            Add Artwork
          </Link>
        </li>
        {pathname === "/add-artwork" ? null : (
          <li className="navbar__list-item">
            <form
              className="navbar__list__search"
              onSubmit={handleSearchSubmit}
            >
              <label htmlFor="search">{}</label>
              <input
                placeholder="Company or Product code"
                className="navbar__list__search-text"
                type="search"
                id="search"
                value={searchQuery}
                onChange={handleSearchInput}
              />
              <label htmlFor="search-button">{}</label>
              <button
                type="submit"
                className="navbar__list__search-button"
                id="search-button"
              >
                <FontAwesomeIcon icon={faMagnifyingGlass} />
              </button>
            </form>
          </li>
        )}
        <li className="navbar__list-item">
          <button
            onClick={signOut}
            type="button"
            className="navbar__list__log-out"
          >
            Log Out
          </button>
        </li>
      </ul>
    </nav>
  );
};

export default NavBar;
