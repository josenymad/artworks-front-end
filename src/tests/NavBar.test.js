import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

it("renders correctly", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <NavBar />
    </BrowserRouter>,
  );

  expect(asFragment()).toMatchSnapshot();
});

it("renders All Artworks", () => {
  render(
    <BrowserRouter>
      <NavBar />
    </BrowserRouter>,
  );
  const allArtworks = screen.getByText("All Artworks");

  expect(allArtworks).toBeInTheDocument();
});

it("renders Add Artwork", () => {
  render(
    <BrowserRouter>
      <NavBar />
    </BrowserRouter>,
  );
  const addArtwork = screen.getByText("Add Artwork");

  expect(addArtwork).toBeInTheDocument();
});
