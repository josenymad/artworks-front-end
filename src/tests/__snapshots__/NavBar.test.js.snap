// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <ul
      class="navbar__list"
    >
      <li
        class="navbar__list__all-artworks"
      >
        <a
          class="navbar__list__all-artworks__link"
          href="/"
        >
          All Artworks
        </a>
      </li>
      <li>
        <a
          class="navbar__list__all-artworks__link"
          href="/add-artwork"
        >
          Add Artwork
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
